### DISCUSSION TITLE
x-show doesn't work with dialog

### DISCUSSION THREAD

**Discussion by florian-sun** [alpinejs/alpine#4386] [7. Bugs (archived)]
The following html shd display the text "Something" but it doesn't

```
<body>   
    <div x-data="{ open: true }">
        <dialog x-show="open">
            <div>
                Something
           </div>
        </dialog>
     </div>
</body>
```

**Comment by SimoTod** [comment]
That's not the way a dialog element works: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

X-show just applies "display: none" when hidden. You must use the native api to control a dialog element.

```html
<div x-data="{ open: true }">
  <button @click="open = !open">toggle</button>
  <dialog x-effect="open ? $el.show() : $el.close()">
    <div>Something</div>
  </dialog>
</div>
```

**Reply by florian-sun** [reply]
Thanks but this just confirms that x-show doesn't work with dialog.
But actually I'm not sure about this. You can still control a dialog with x-show **if** the dialog is visible using the regular "open" attribute.
Using the following code you can toggle the dialog using x-show:

```
<body>   
    <div x-data="{ open1: true }">
        <button @click="open1 = ! open1">Toggle</button>
        <dialog x-show="open1" open>
            <div>
                Something
            </div>
        </dialog>
    </div>
</body>
```

**Reply by ekwoka** [reply]
Sure, but then you're kind of hacking it.

at that point just `:open=open1` would get a better result.

or `x-effect="open1 ? $el.show() : $el.close()"`

Maybe what you really want is a `detail` element, anyway?

But x-show is not meant to work with dialogs or details.
